src/stories/components/Form/Form.js

```diff
@@ -0,0 +1,211 @@
+        case 'login':
+            return <Login />;
+        case 'signup':
+            return <SignUp />;
+        default:
+            return <CustomForm elements={formItems}/>;
+        }
+    }
+    return (
+        <form>
+            <div className="form__container">
+                {returnForm(type)}
+                <div className="form__button">
+                    <Button
+                        modifierClasses="button--quaternary button--small"
+                        text={(buttonText) ?
+                            buttonText :
+                            (type=='login') ?
+                                'Login':
+                                (type=='signup') ?
+                                    'Sign Up':
+                                    'Submit'}
+                        onClick={handleSubmit}
+                    />
+                </div>
+            </div>
+        </form>
+    );
+};
+
+Form.propTypes = {
+    /**
+   * Form's list of form items
+   */
+    formItems: PropTypes.array,
+    /**
+   * Form's type
+   */
+    type: PropTypes.string.isRequired,
+    /**
+     * Form's button text
+     */
+    buttonText: PropTypes.string,
+    /**
+     * Form's button action
+     */
+    handleSubmit: PropTypes.func,
+};
+
+Form.defaultProps = {
+    formItems: [],
+    type: 'text',
+    buttonText: 'Submit',
+    handleSubmit: null,
+};
+
+/**
+ * Render <CustomForm /> HTML
+ * @param {array} elements list of form items to render.
+ * @return {object} (
+ *   <Login />
+ * )
+ */
+const CustomForm = ({ elements }) => (
+    <React.Fragment>
+        {
+            elements.map((element, index)=> (
+                <div
+                    className="form__item"
+                    key={index}>
+                    <FormItem
+                        type={element.type}
+                        placeholder={element.placeholder}
+                        showLabel = {element.showLabel}
+                        options = {element.options}
+                        value={element.value}
+                        label={element.label}
+                        modifierClasses={element.modifierClasses}
+                    />
+                </div>
+            ))
+        }
+    </React.Fragment>
+);
+
+CustomForm.propTypes = {
+    /**
+    * CustomForm's allowed types
+   */
+    elements: PropTypes.arrayOf(PropTypes.shape({
+        type: PropTypes.string.isRequired,
+        label: PropTypes.string,
+        placeholder: PropTypes.string,
+        value: PropTypes.string,
+        options: PropTypes.array,
+        showLabel: PropTypes.bool,
+        modifierClasses: PropTypes.string,
+    })).isRequired,
+    /**
+   * CustomForm's type
+   */
+    type: PropTypes.string,
+};
+
+CustomForm.defaultProps = {
+    type: 'custom',
+};
+
+/**
+ * Render <Login /> HTML
+ * @return {object} (
+ *   <Login />
+ * )
+ */
+const Login = () => (
+    <React.Fragment>
+        <div className="form__item">
```

Comment:
You don't need these wrapper `div`'s. Form items already have a wrapper around them, if you need it for styling, feel free to update margins, spacing or whatever else here: https://github.com/neu-self-image-experiments/sie-firebase/blob/master/src/stories/components/FormItem/styles.scss

Reply:
I need the div to stretch the form element to the full-width of the container using `flex-basis: 100%`, do you suggest I make this change in the `<FormItem />` component itself?

Reply:
Figured it out, too many div wrappers were causing the issue